### PR TITLE
Data docs

### PR DESCRIPTION
**Pull request by levkk**
### Features

1. Fork `@codemirror/lang-sql` to add support for Postgres-specific syntax.
2. Change docs text color to white.
3. Markdown tables in docs are now `.table-sm` by default instead of the ugly default bootstrap tables.
4. Add anchor (link) support for headings/table of contents markdown parser.
5. Add padding to the bottom of the code block to offset horizontal scrolling bar.

### Documentation

1. Documented logical replication, FDWs, and COPY in data import.

### CODE REVIEW

File: pgml-dashboard/src/utils/markdown.rs [hunk 1]
```diff
@@ -513,15 +513,34 @@ pub fn get_toc<'a>(root: &'a AstNode<'a>) -> anyhow::Result<Vec<TocLink>> {
                         return Ok(false);
                     }
                 };
-                if let NodeValue::Text(text) = &sibling.data.borrow().value {
-                    let index = match header_count.get(text) {
+
+                println!("child: {:?}", sibling);
```
Comment (chillenberger) [inline pgml-dashboard/src/utils/markdown.rs:517]: Do you want this println in here?

File: pgml-cms/docs/SUMMARY.md [hunk 2]
```diff
@@ -50,7 +51,7 @@
 
 ## Product
 
-* [Cloud Database](product/cloud-database/README.md)
```
Comment (montanalow) [inline pgml-cms/docs/SUMMARY.md:53]: This is Cloud as opposed to Vector section

File: pgml-cms/docs/SUMMARY.md [hunk 3]
```diff
@@ -79,7 +80,7 @@
 ## Resources
 
 * [FAQs](resources/faqs.md)
-* [Data Storage & Retrieval](resources/data-storage-and-retrieval/README.md)
+* [Data Storage & Retrieval](resources/data-storage-and-retrieval/tabular-data.md)
```
Comment (montanalow) [inline pgml-cms/docs/SUMMARY.md:83]: This is a dup?

File: pgml-cms/docs/introduction/getting-started/README.md [hunk 4]
```diff
@@ -4,11 +4,11 @@ description: Setup a database and connect your application to PostgresML
 
 # Getting Started
 
-A PostgresML deployment consists of multiple components working in concert to provide a complete Machine Learning platform. We provide a fully managed solution in our cloud.
+A PostgresML deployment consists of multiple components working in concert to provide a complete Machine Learning platform. We provide a fully managed solution in our cloud, and document a self-hosted installation in our docs.
```
Comment (montanalow) [inline pgml-cms/docs/introduction/getting-started/README.md:7]: documenting the self hosted version is confusing here. Are these docs for cloud users, or self hosted people?

File: pgml-cms/docs/introduction/getting-started/README.md [hunk 5]
```diff
@@ -4,11 +4,11 @@ description: Setup a database and connect your application to PostgresML
...
 
-* A PostgreSQL database, with pgml and pgvector extensions installed, including backups, metrics, logs, replicas and high availability configurations
-* A PgCat pooling proxy to provide secure access and model load balancing across tens of thousands of clients
-* A web application to manage deployed models and host SQL notebooks
+* PostgreSQL database, with `pgml`, `pgvector` and many other extensions installed, including backups, metrics, logs, replicas and high availability
+* PgCat pooler to provide secure access and model load balancing across thousands of clients
+* A web application to manage deployed models and write experiments in SQL notebooks
```
Comment (montanalow) [inline pgml-cms/docs/introduction/getting-started/README.md:11]:
```suggestion
* A web application to manage deployed models and share experiments and analysis in SQL notebooks
```

File: pgml-cms/docs/introduction/getting-started/connect-your-app.md [hunk 6]
```diff
@@ -4,13 +4,13 @@ description: PostgresML is compatible with all standard PostgreSQL clients
 
 # Connect your app
 
-You can connect to your database from any Postgres compatible client. PostgresML is intended to serve in the traditional role of an application database, along with it's extended role as an MLOps platform to make it easy to build and maintain AI applications.
+You can connect to your database from any PostgresSQL-compatible client. PostgresML is intended to serve in the traditional role of an application database, along with it's extended role as an MLOps platform to make it easy to build and maintain AI applications together with your application data.
 
-## Application SDKs
+## SDK
```
Comment (montanalow) [inline pgml-cms/docs/introduction/getting-started/connect-your-app.md:9]:
```suggestion
## Client SDK
```

We've waffled between singular and plural. Changing this also means changing urls for consistency.

File: pgml-cms/docs/introduction/getting-started/connect-your-app.md [hunk 7]
```diff
@@ -4,13 +4,13 @@ description: PostgresML is compatible with all standard PostgreSQL clients
...
 
-We provide client SDKs for JavaScript, Python and Rust apps that manage connections to the Postgres database and make it easy to construct efficient queries for AI use cases, like managing a document collection for RAG, or building a chatbot. All of the ML & AI still happens in the database, with centralized operations, hardware and dependency management.
+We provide a client SDK for JavaScript, Python and Rust. The SDK manages connections to the Postgres database and makes it easy to construct efficient queries for AI use cases, like managing a document collection for RAG, or building a chatbot. All of the ML & AI still happenening inside the database, with centralized operations, hardware and dependency management.
```
Comment (montanalow) [inline pgml-cms/docs/introduction/getting-started/connect-your-app.md:11]:
```suggestion
We provide a client SDK for JavaScript, Python and Rust. The SDK manages connections to the Postgres database and makes it easy to construct efficient queries for AI use cases, like managing a document collection for RAG, or building a chatbot. All of the ML & AI still happens inside the database, with centralized operations, hardware and dependency management.
```

File: pgml-cms/docs/introduction/getting-started/connect-your-app.md [hunk 8]
```diff
@@ -4,13 +4,13 @@ description: PostgresML is compatible with all standard PostgreSQL clients
...
 
-These SDKs are under rapid development to add new features and use cases, but we release non breaking changes with minor version updates in accordance with SemVer. It's easy to install into your existing application.
+The SDK are under rapid development to add new features and use cases, but we release non breaking changes with minor version updates in accordance with SemVer. It's easy to install into your existing application.
```
Comment (montanalow) [inline pgml-cms/docs/introduction/getting-started/connect-your-app.md:13]:
```suggestion
The SDK is under rapid development to add new features and use cases, but we release non breaking changes with minor version updates in accordance with SemVer. It's easy to install into your existing application.
```

File: pgml-cms/docs/introduction/getting-started/connect-your-app.md [hunk 9]
```diff
@@ -28,6 +28,8 @@ pip install pgml
 {% endtab %}
 {% endtabs %}
 
+Our SDK comes with zero additional dependencies. The core of the SDK is written in Rust, and we provide language bindings and native packaging & distribution.
```
Comment (montanalow) [inline pgml-cms/docs/introduction/getting-started/connect-your-app.md:31]:
```suggestion
Our SDK comes with zero additional dependencies, to provide the simplest and safest ML application deployment and maintenance possible. The core of the SDK is written in Rust, and we provide language bindings and native packaging & distribution.
```

File: pgml-cms/docs/introduction/getting-started/connect-your-app.md [hunk 10]
```diff
@@ -101,9 +103,9 @@ You can also connect directly to the database with your favorite bindings or ORM
 * Rust: [postgres](https://crates.io/crates/postgres), [SQLx](https://github.com/launchbadge/sqlx) or [Diesel](https://github.com/diesel-rs/diesel)
 * Swift: [PostgresNIO](https://github.com/vapor/postgres-nio) or [PostgresClientKit](https://github.com/codewinsdotcom/PostgresClientKit)
 
-## SQL Editors
+## SQL editors
 
-Use any of these popular tools to execute SQL queries directly against the database:
+If you need to write ad-hoc queries, you can use any of these popular tools to execute SQL queries directly on your database:
```
Comment (montanalow) [inline pgml-cms/docs/introduction/getting-started/connect-your-app.md:108]:
```suggestion
If you need to write ad-hoc queries, or perform administrative functions, you can use any of these popular tools to execute SQL queries directly on your database:
```

File: pgml-cms/docs/introduction/getting-started/import-your-data/README.md [hunk 11]
```diff
@@ -1,22 +1,26 @@
...
+AI needs data, whether it's generating text with LLMs, creating embeddings, or training regression or classification models on customer data.
 
-PostgresML can easily ingest data from your existing data stores.
+Just like any PostgreSQL database, PostgresML can be configured as the primary application database, a logical replica of your primary database, or with foreign data wrappers to query your primary database on demand. Depending on how frequently your data changes and your latency requirements, one approach is better than the other.
 
-## Static data
+## Primary database
 
-Data that changes infrequently can be easily imported into PostgresML using `COPY`. All you have to do is export your data as a CSV file, create a table in Postgres to store it, and import it using the command line.
+If you're intention is to use PostgresML as your primary database, your job here is done. You can use the connection credentials provided and start building your application on top of in-database AI right away.
```
Comment (montanalow) [inline pgml-cms/docs/introduction/getting-started/import-your-data/README.md:9]:
```suggestion
If your intention is to use PostgresML as your primary database, your job here is done. You can use the connection credentials provided and start building your application on top of in-database AI right away.
```

File: pgml-dashboard/src/utils/markdown.rs (hunk 1 above)
Comment (montanalow) [inline pgml-dashboard/src/utils/markdown.rs:517]:
```suggestion
```

**Review by montanalow — COMMENTED** [review]
Nice!

File: pgml-cms/docs/SUMMARY.md [hunk 12]
```diff
@@ -79,7 +80,7 @@
...
   * [Tabular data](resources/data-storage-and-retrieval/tabular-data.md)
```
Comment (montanalow) [inline pgml-cms/docs/SUMMARY.md:84]:
```suggestion
```

**Review by chillenberger — COMMENTED** [review]
👍

File: pgml-dashboard/src/utils/markdown.rs (hunk 1 above)
Comment (levkk) [reply pgml-dashboard/src/utils/markdown.rs:517]: I do not :) good catch

File: pgml-cms/docs/SUMMARY.md (hunk 3 above)
Comment (levkk) [reply pgml-cms/docs/SUMMARY.md:83]: Not a dup, README.md is currently empty.

File: pgml-cms/docs/introduction/getting-started/README.md (hunk 4 above)
Comment (levkk) [reply pgml-cms/docs/introduction/getting-started/README.md:7]: Both.

File: pgml-cms/docs/introduction/getting-started/connect-your-app.md (hunk 6 above)
Comment (levkk) [reply pgml-cms/docs/introduction/getting-started/connect-your-app.md:9]: Not going to write Client SDKs ever again, that wording is just wrong. SDK are only usable on the client by definition.

Comment (levkk) [reply pgml-cms/docs/introduction/getting-started/connect-your-app.md:9]: It's going to be singular going forward.

**Pull request merged**